src/mon/NVMeofGwMon.cc

```diff
@@ -115,18 +115,25 @@ void NVMeofGwMon::tick()
       _propose_pending |= propose;
       last_beacon.erase(lb);
     } else {
-      BeaconSubsystems  *subsystems =
-         &pending_map.created_gws[lb.group_key][lb.gw_id].subsystems;
-      if (subsystems && subsystems->size() && old_group_key != lb.group_key) {
-        // to call track_deleting_gws once per each group-key
-        pending_map.track_deleting_gws(lb.group_key, *subsystems, propose);
-        old_group_key = lb.group_key;
-        _propose_pending |= propose;
-      }
       dout(20) << "beacon live for GW key: " << lb.gw_id << dendl;
     }
   }
-
+  BeaconSubsystems   empty_subsystems;
+  for (auto& group_state: pending_map.created_gws) {
+    auto& group_key = group_state.first;
+    auto& gws_states = group_state.second;
+    BeaconSubsystems  * subsystems = &empty_subsystems;
+    for (auto& gw_state : gws_states) { // loop for GWs inside nqn group
+      auto& gw_id = gw_state.first;
+      subsystems =
+        &pending_map.created_gws[group_key][gw_id].subsystems;
+      if (subsystems->size()) { // Set subsystems to the valid value
+        break;
+      }
+    }
+    pending_map.track_deleting_gws(group_key, *subsystems, propose);
```

Comment:
I don't think I understand the logic here.  For each group, this will end up being invoked on a) the subsystems set first gw_id with a non-empty subsystems or b) that of the highest gw_id even if empty.  Is that the intent?  If so, it's worth explaining why that is the desired behavior.

Reply:
Sorry, explaining as a comment in the code.  If I'm confused, likely future readers will be as well.